app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -200,11 +200,16 @@ def miq_templates
       hardware.size_on_disk = i.size
       hardware.virtualization_type = i.properties.try(:[], 'hypervisor_type') || i.attributes['hypervisor_type']
 
+      vm = VmOrTemplate.find_by(:ems_ref => i.id)
+      os_name = vm&.operating_system&.product_name
```

Comment:
I'm not sure if this kind of lookup in refresh is okay, but let @mansam review!

Reply:
This looks like it should be OK, it's one simple query per vm or image. @Ladas, do you have any opinions on this approach?

Reply:
Hm, we usually don't do this, since there can be hundreds of thousands of vms or images.

Btw. I think the approach should be another way. Refresh and smartstate should not be changing the same attributes, we should remove it from one or the other. Right now, refresh would keep rewriting what was set by smartstate.

Reply:
Refresh and smartstate have always fought over this column,  because for different sources we may or may not get one or the other.  In the classic refresh, one of them was chosen as a winner (I believe smartstate always wins, but I'll check and find a link).

Reply:
Oh, and we should not be querying in the collector code like this, especially once-per anything as it introduces nasty N+1 performance issues.

Reply:
Ok, thank you for comments. I will try to find better solution for this issue ASAP.

Reply:
so we should be storing only ```operating_system.product_name = guest_os``` unless guest_os is unknown

Since it uses ```OperatingSystem.normalize_os_name(i.try(:os_distro) || 'unknown')```, smartstate should be storing there the same normalized value. And we could keep the rest for the smatstate. This approach should solve the BZ?

@Fryguy graph refresh doesn;t do any of ^ examples, since all of them are N+1 and going forward, collector will not able to even reach the DB. So best would be to have different schemas.

Reply:
@Ladas we could check the Vm's `last_scan_on` to see if smartstate has been run for the Vm (we can do this direct to the DB for MIQ or using the API in the future for topological) and not set the attribute in the parser so as to not overwrite the value smartstate sets.

Reply:
Well..I have deleted lines responsible for setting the OS. Now, there are no changes in OS information after refresh.

Reply:
@evgeny-chufarov well we still want to set the OS info if smartstate wasn't run.

Reply:
Not collecting it feels like the wrong approach as well.  SmartState is an opt-in thing, whereas refresh is not.  I would expect refresh to always collect at least something.

Reply:
@Fryguy @agrare Am I understanding properly that the recommended course of action here is for us to query the DB once at collection time for all the instances associated with the manager, and then compare their last scan time?

Reply:
@mansam yes that's correct

Reply:
@aufi 
@mansam 
Now we are querying the database for information about the **last_scan time** of Smart State Analysis. If it is, the refresh skips this vm/template and checks the others. If no SSA scans were executed yet, refresh will try to collect any info that it can.